sdk/python/kfp/dsl/_container_op.py

```diff
@@ -16,11 +16,125 @@
 from . import _pipeline
 from . import _pipeline_param
 import re
-from typing import Dict
-
+from typing import Dict, List
+from abc import ABCMeta, abstractmethod
+from . import _types
+
+class BaseMeta(object):
+  __metaclass__ = ABCMeta
+  def __init__(self):
+    pass
+
+  @abstractmethod
+  def to_dict(self):
+    pass
+
+  def serialize(self):
+    import yaml
+    return yaml.dump(self.to_dict())
+
+class TypeMeta(BaseMeta):
+  def __init__(self, name, properties):
+    self.name = name
+    self.properties = properties
+
+  def to_dict(self):
+    return {self.name: self.properties}
+
+class ParameterMeta(BaseMeta):
```

Comment:
Can you please use kfp.components._structures.InputSpec for this?
It's the same structure created for the same purpose.